src/Utility/Text.php

```diff
@@ -1170,16 +1170,18 @@ public static function slug($string, $options = [])
             $string = static::transliterate($string, $options['transliteratorId']);
         }
 
-        $regex = '^\s\p{Ll}\p{Lm}\p{Lo}\p{Lt}\p{Lu}\p{Nd}';
+        $regex = '^\p{Ll}\p{Lm}\p{Lo}\p{Lt}\p{Lu}\p{Nd}';
         if ($options['preserve']) {
             $regex .= preg_quote($options['preserve'], '/');
         }
         $quotedReplacement = preg_quote($options['replacement'], '/');
         $map = [
-            '/[' . $regex . ']/mu' => ' ',
-            '/[\s]+/mu' => $options['replacement'],
+            '/[' . $regex . ']/mu' => $options['replacement'],
             sprintf('/^[%s]+|[%s]+$/', $quotedReplacement, $quotedReplacement) => '',
         ];
+        if ($options['replacement']) {
+            $map[sprintf('/[%s]+/mu', $quotedReplacement)] = $options['replacement'];
```

Comment:
Why not just change the above pattern?  Also if this is checking for empty string can you use ===?

Reply:
I'm not sure what you mean by change the above pattern. This pattern replaces duplicates "replacements" with a single one after the first regexp.

I was under the impression we generally used loose checks for "empty" but I'm happy to add the strict equality check instead.

Reply:
You could use `isset()`

Reply:
I don't think that will work for an empty string though.

Reply:
It wouldn't you would have to do `isset() && strlen()`.

Reply:
The key is assumed to exist elsewhere with a default.  I think just `=== ''` works.

Reply:
Before these changes, `null` and `false` were valid replacement options (or at least the code didn't complain, just replaced with nothing) and this change is BC to that. I added test cases (which pass in master too) for those situations.

I'm fine with `isset() && strlen()`.

Reply:
The doc states `- replacement: Replacement string. Default '-'.`  So see no need to to account for `replacement` being set to `null` and hence `isset()` check is not necessary.